### PR TITLE
Specs v2 and v3

### PR DESCRIPTION
**Pull request by cyberbikepunk**
cc @akariv @annafralberts @seegerseeger

### CODE REVIEW

**Comment by cyberbikepunk** [comment]
@pwalsh got a 404?. I thought i was cleaner to branch this off master.

**Comment by cyberbikepunk** [comment]
Hey @akariv @pwalsh @pudo @seegerseeger @annafralberts There are things to take into consideration for getting this right. Here are my thoughts...

## File format:

YAML has been preferred over JSON because it supports comments and the entire things is written like a tutorial. However, processing the hand written files that are trickling down shows that @pwalsh's humble opinion about whitespace is not far from the truth. the question is kinda: do you wanna spend your time answering questions about how to fill the doc or do you prefer fixing white-space errors? I haven't given up on YAML yet because at this stage I haven't hammered that point to the contributors and maybe the situation can improve. @akariv 's suggestion of using an online web-app is promising.

## Version 1

So far we've been working with a much simpler file, where the contributor could ad-lib on the fields (s)he found on the data. The idea is that the user suggested the mapping by providing a translation and a  description of the field. I got very little of the latter, which is really what is supposed to help out the person writing the transform code downstream. If that person has to do the research again to find out how the fields map, we're losing.

## version 2

now we're basically suggesting all the columns directly inside the template. this makes it much easier for the user to fill in. The drawback is that  we're expecting a 1-to-1 mapping between columns in the raw data and those in the final fiscal package. I deliberately ignored corner cases but this probably was a mistake. e.g. I but hadn't anticipated a case like Austria:

```
        - name: Status G = genehmigt, A = ausbezahlt
          translation: G = commited, A = paid out
          type: string
          missingValue:
          description:
```

In  the above example, the we're expecting the columns `committed` and `paid out` and instead we're getting a single "measure" column binded to a "category" column. This is like trying to shove a cube through a round hole.

## Why not a datapackage format directly?

I was really putting myself in the shoes of the contributor here. This extra-step makes it more accessible and less error prone.
1. slugify the title to get the name (not that important)
2. supporting multiple identical resources with the same schema to avoid unmanageably long files (more-useful: there are many cases of data sets broken down into multiple files (error prone in my -humble- opinion)... maybe it would be nice if the datapackage specs could support that?.

## conclusion

 i'm thinking that the looser approach of version 1, combined with emphasis on contributor training on how to do proper field descriptions, combined with an optional web app like [this](http://exavolt.github.io/onde/) could be the winner.

What do you think? I would appreciate feedback in the meantime, I'm moving on to modeling the fiscal package now and pushing some of the datasets up to os.

**Comment by pwalsh** [comment]
Hi!

My opinion about YAML and non-technies is not humble - I was just being polite :). 

YAML is a horrible format for _writing_ when the writer does not understand simple concepts a programmer takes for granted - like 3 spaces before a `-` actually having a completely different meaning to 2 spaces.

@cyberbikepunk I'm in complete agreement with @akariv that a web form is a much better solution, and if we were brought in to the conversation earlier, we could have helped shape this decision.

However, due to us having time constraints here, we need to take a wholistic view - will we lose momentum by shifting midstream? Is it quicker, for the current sourcing exercise, to just keep going and fix errors? My suggestion is that when @akariv gets back to work on Monday that you consult with him on this.

**Comment by cyberbikepunk** [comment]
@pwalsh @akariv I'm thinking we should pursue as is. I can live with it. I just need to create the underlying schema and suggest the interface as an option. Although an interface without the possibility of adding guidelines is not ideal.

**Comment by cyberbikepunk** [comment]
@akariv basically concerning `source.description.yaml`:
- v1 has been used up to now: will be support to avoid having to re-write everything
- v2 was a bad idea
- v3 is a better version of v1 and should be used from now on.

File: data/AT.austria/AT11.burgenland/pipeline-specs.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+AT11.burgenland:
+  pipeline: {params: source.datapackage.json, run: validate_datapackage}
+  schedule: {crontab: '@annually'}
```
Comment (akariv) [inline data/AT.austria/AT11.burgenland/pipeline-specs.yaml:3]: this syntax won't work - you need the traditional crontab format

File: data/AT.austria/AT11.burgenland/pipeline-specs.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+AT11.burgenland:
```
Comment (akariv) [inline data/AT.austria/AT11.burgenland/pipeline-specs.yaml:1]: Why are pipeline-specs in git anyway? Aren't they supposed to be generated automatically?

File: data/PL.poland/validate_tabular_remote_source.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+""" A processor to download a data-file from a link and validate it."""
```
Comment (akariv) [inline data/PL.poland/validate_tabular_remote_source.py:1]: I guess that this is still a work in progress?

What is the purpose of this code?

Comment (cyberbikepunk) [reply data/PL.poland/validate_tabular_remote_source.py:1]: junk

**Pull request merged**